### PR TITLE
fix: alert permissions for api scope

### PR DESCRIPTION
**Pull request by gaetanmaisse**
Replace https://github.com/gravitee-io/old-gravitee-api-management/pull/370

**Issue**

https://github.com/gravitee-io/issues/issues/5974

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/issues-5974-alert-permissions/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

### CODE REVIEW

File: gravitee-apim-console-webui/src/components/alerts/alert/alert.html [hunk 1]
```diff
@@ -214,17 +221,21 @@ <h3>Condition</h3>
...
 
           <md-tab md-on-select="$ctrl.currentTab=$ctrl.selectTab(1)">
             <md-tab-label>Notifications</md-tab-label>
             <md-tab-body>
               <!-- Dampening -->
-              <gv-alert-trigger-dampening dampening="$ctrl.alert.dampening"></gv-alert-trigger-dampening>
+              <gv-alert-trigger-dampening dampening="$ctrl.alert.dampening" is-readonly="$ctrl.isReadonly()"></gv-alert-trigger-dampening>
 
-              <gv-alert-notifications alert="$ctrl.alert"></gv-alert-notifications>
+              <gv-alert-notifications alert="$ctrl.alert" is-readonly="$ctrl.isReadonly()" yann="true"></gv-alert-notifications>
```
Comment (marcambier) [inline gravitee-apim-console-webui/src/components/alerts/alert/alert.html:238]:
```suggestion
              <gv-alert-notifications alert="$ctrl.alert" is-readonly="$ctrl.isReadonly()"></gv-alert-notifications>
```

File: gravitee-apim-console-webui/src/components/alerts/alert/alert.component.ts [hunk 2]
```diff
@@ -171,6 +173,20 @@ const AlertComponent: ng.IComponentOptions = {
         $state.go('management.settings.alerts.list');
       }
     };
+
+    this.hasPermissionForCurrentScope = (permission: string): boolean => {
```
Comment (marcambier) [inline gravitee-apim-console-webui/src/components/alerts/alert/alert.component.ts:177]: Suggestion : would it be relevant to put this function in user.service, instead of duplicating this code ? (Or another relevant service ?)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-container/pom.xml [hunk 3]
```diff
@@ -145,6 +145,12 @@
             <artifactId>jul-to-slf4j</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (marcambier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-container/pom.xml:148]: Is that dependency addition related to this feature ?

**Review by marcambier — COMMENTED** [review]
Made 3 comments on this PR.

And 1 more here : Is this file rename OK ? plugins/.gitignore → plugins_old/.gitignore 

Code LGTM, i'll test in details.

File: gravitee-apim-console-webui/src/components/alerts/alert/alert.component.ts (hunk 2 above)
Comment (ytvnr) [reply gravitee-apim-console-webui/src/components/alerts/alert/alert.component.ts:177]: to be honest, I wanted to refactor it to have it in a alert helper (for example)
The problem with that is we need the $state to get the apiId or appId param and be able to know which scope to choose.
as discussed with @gaetanmaisse, we prefer to keep the $state info inside this component, instead of injecting it in a new class. I don't know if passing it as a param would resolve the thing
wdyt ?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-container/pom.xml (hunk 3 above)
Comment (ytvnr) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-container/pom.xml:148]: nop, I add some issues running 3.5.x in ee mode and I unfortunately commit :)

File: gravitee-apim-console-webui/src/components/alerts/alert/alert.component.ts (hunk 2 above)
Comment (marcambier) [reply gravitee-apim-console-webui/src/components/alerts/alert/alert.component.ts:177]: it's ok to me

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/issues-5974-alert-permissions/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/issues-5974-alert-permissions/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.

**Review by marcambier — APPROVED** [review]

**Comment by graviteeio** [comment]
**Gravitee.io Automatic Deployment**

Congrats, all the CI steps were 🟢 . The Bot was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/issues-5974-alert-permissions/index.html)! 🚀

_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.
